### PR TITLE
Adding recipe for mappyfile

### PR DESCRIPTION
**Pull request by tchaddad**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tchaddad** [comment]
@conda-forge/help-python this recipe for mappyfile has passed all checks and is ready for your review. Thank you!

File: recipes/mappyfile/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+requirements:
+  host:
+    - pip
+    - python >=3.8
+  run:
+    - click   
+    - jsonref ==0.2
+    - jsonschema >=2.0   
+    - lark-parser >=0.11.3
```
Comment (synapticarbors) [inline recipes/mappyfile/meta.yaml:28]: Upstream in the `setup.py` the requirement is: `lark>=1.1.5`. Please check to reconcile possible confusion between `lark` and `lark-parser`.

**Comment by geographika** [comment]
@tchaddad - the current master branch of mappyfile is moving to Python3 and so the requirements have been updated since the 0.9.7 release, which supports Python2. 

As this recipe is for the current 0.9.7 release then I think the requirements should match those in https://github.com/geographika/mappyfile/blob/v0.9.7/setup.py

```
'lark-parser>=0.11.3',
'pyrsistent<0.17.0; python_version=="2.7"',
'jsonschema>=2.0, <=3.2.0; python_version=="2.7"',
'jsonschema>=2.0; python_version>="3.0.0"',
'jsonref==0.2',
'click; python_version>="3.0.0"',
'click < 8.0.0; python_version=="2.7"'
```

I'm not sure of the syntax, but I think it should look something like:

```
requirements:
  run:
    - lark-parser >=0.11.3

    # Dependencies for Python 2.7
    - pyrsistent <0.17.0; python_version == "2.7"
    - jsonschema >=2.0, <=3.2.0; python_version == "2.7"

    # Dependencies for Python 3.x
    - jsonschema >=2.0; python_version >= "3.0.0"
    - jsonref ==0.2

    # Common dependency for Python 3.x
    - click; python_version >= "3.0.0"

    # Common dependency for Python 2.7
    - click <8.0.0; python_version == "2.7"
```

The Python requirement `python >=3.8` can also be changed to `python` (I think) to allow 2.7 support.

File: recipes/mappyfile/meta.yaml (hunk 1 above)
Comment (geographika) [reply recipes/mappyfile/meta.yaml:28]: @synapticarbors thanks for reviewing this. I've clarified the requirements in the comments.

**Comment by beenje** [comment]
Recipe looks good to me.

@geographika could you please confirm you want to be listed as maintainer?

**Comment by synapticarbors** [comment]
When an update is detected on pypi, a conda-forge bot will automatically create a PR with the necessary changes. Usually it's just a matter of checking the dependencies versions for changes and that the CI passes.

File: recipes/mappyfile/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - setuptools
+    - python >=3.8
+
+test:
```
Comment (synapticarbors) [inline recipes/mappyfile/meta.yaml:32]: You might consider adding a `pip check` here:

https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L58-L64

**Pull request merged**